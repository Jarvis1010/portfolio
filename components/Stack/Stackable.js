import styled from "styled-components";

const type = (theme = {}) => {
  const defaultTheme = {
    mediumGrey: "lightgrey",
    darkGrey: "grey",
    black: "black",
    white: "white"
  };
  const finalTheme = { ...defaultTheme, ...theme };
  return {
    light: { "background-color": finalTheme.white, color: finalTheme.black },
    medium: {
      "background-color": finalTheme.mediumGrey,
      color: finalTheme.black
    },
    dark: { "background-color": finalTheme.darkGrey, color: finalTheme.white },
    transparent: { "background-color": "transparent", color: finalTheme.white },
    default: { "background-color": finalTheme.white, color: finalTheme.black }
  };
};

const Stack = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  ${props =>
    Object.entries(type(props.theme)[props.type] || type().default)
      .map(x => x.join(":"))
      .join(";")};
`;

export default Stack;
