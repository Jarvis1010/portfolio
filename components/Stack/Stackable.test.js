import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Stackable from "./Stackable";

describe("Stackable Tests", () => {
  it("renders with proper gaps", () => {
    ["light", "medium", "dark", null].forEach(x => {
      const wrapper = shallow(<Stackable type={x} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
