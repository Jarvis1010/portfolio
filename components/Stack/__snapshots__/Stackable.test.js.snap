// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stackable Tests renders with proper gaps 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  width: 100%;
  height: 100%;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  type="light"
/>
`;

exports[`Stackable Tests renders with proper gaps 2`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  width: 100%;
  height: 100%;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  type="medium"
/>
`;

exports[`Stackable Tests renders with proper gaps 3`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  width: 100%;
  height: 100%;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  type="dark"
/>
`;

exports[`Stackable Tests renders with proper gaps 4`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  width: 100%;
  height: 100%;
  ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  type={null}
/>
`;
